### ISSUE TITLE
Image preview shows incorrect canvas size when editor is scaled down vertically 

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#81878]
**Repo**
1. Have an integrated terminal open below editors
1. Open an image preview in main editor area
1. Make sure its zoom is set to `whole image`
1. Now expand the integrated terminal upwards (shrinking the image editor)

**Bug**
The canvas size indicators (the checkerboard pattern) ends up growing in width

![Screen Shot 2019-10-02 at 4 33 46 PM](https://user-images.githubusercontent.com/12821956/66089074-91dfce80-e532-11e9-90a0-6b144e239089.png)

**Comment by mjbvz** [comment]
Here's the relevant css: https://github.com/microsoft/vscode-distro/blob/b058a2831a30cf71e8f0d216cf841071eb3db957/extensions/image-preview/media/main.css#L5

**Comment by Sti2nd** [comment]
> The canvas size indicators (the checkerboard pattern) ends up growing in width

I cannot reproduce that it grows in width, but I can reproduce that it maintains the same width as the image before expanding the terminal. 

I have looked at the CSS file and parsed DOM, essentially you end up with an img-element that has this CSS (I removed some properties):
```CSS
img {
  background-image:
    linear-gradient(45deg, rgb(20, 20, 20) 25%, transparent 25%, transparent 75%, 
      rgb(20, 20, 20) 75%, rgb(20, 20, 20)),
    linear-gradient(45deg, rgb(20, 20, 20) 25%, transparent 25%, transparent 75%, 
      rgb(20, 20, 20) 75%, rgb(20, 20, 20));
  background-size: 16px 16px;
  background-position: 0px 0px, 8px 8px;
  max-width: calc(100% - 20px);
  max-height: calc(100% - 20px);
  object-fit: contain;
}
```
The checkerboard is created by the `background-image` property, and the `object-fit` property is set when zoom is set to `whole image`. The reason we see the checkerboard pattern is because the object is letterboxed when using `object-fit: contain` ([see the link in the contain value on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#Values)).

**Question: ** So now I am wondering if I misunderstood the issue. To me it seems like the checkerboard pattern is supposed to show after expanding terminal because it shows the true size of the img-element. If that wasn't intentional; when should the checkerboard pattern show?

**Comment by Sti2nd** [comment]
I just tried to remove the background-image property and forgot that there are also a border:

```CSS
.container.image img {
  border: 1px solid var(--vscode-imagePreview-border);
}
```

If I remove the border, resizing the image by expanding the terminal starts to lag. No clue why! Maybe it is better to change the border to use the variable `--vscode-editor-background` to work around the lagging?

**Comment by jeanp413** [comment]
I think the checkerboard pattern is used to distinguish the transparent parts of an image, try using an image with alpha channel

**Comment by jeanp413** [comment]
@Sti2nd I fixed it with the following changes. Let me know If you want to create the PR.

```diff
diff --git a/extensions/image-preview/media/main.css b/extensions/image-preview/media/main.css
index 913d2b4672..0f50cef683 100644
--- a/extensions/image-preview/media/main.css
+++ b/extensions/image-preview/media/main.css
@@ -4,8 +4,8 @@
  *--------------------------------------------------------------------------------------------*/
 
 html, body {
+       width: 100%;
        height: 100%;
-       max-height: 100%;
        text-align: center;
 }
 ```

**Comment by mjbvz** [comment]
Fixed by #82027
